private/buf/buflsp/file.go

```diff
@@ -622,16 +622,36 @@ func (f *file) importToSymbol(imp ir.Import) *symbol {
 	}
 }
 
+// messageToSymbols takes an [ir.MessageValue] and returns the symbols parsed from it.
 func (f *file) messageToSymbols(msg ir.MessageValue) []*symbol {
+	return f.messageToSymbolsHelper(msg, 0, nil)
+}
+
+// messageToSymbolsHelper is a recursive helper for extracting the symbols from a [ir.MessageValue].
+func (f *file) messageToSymbolsHelper(msg ir.MessageValue, index int, parents []*symbol) []*symbol {
 	var symbols []*symbol
 	for field := range msg.Fields() {
 		if field.ValueAST().IsZero() {
 			continue
 		}
 		for element := range seq.Values(field.Elements()) {
-			span := element.Value().KeyASTs().At(element.ValueNodeIndex()).Span()
-			elem := &symbol{
-				// NOTE: no [ir.Symbol] for option elements
+			key := element.Value().KeyASTs().At(element.ValueNodeIndex())
```

Comment:
```suggestion
			key := field.KeyASTs().At(element.ValueNodeIndex())
```
The `element.Value()` I think is just the `field` here?